Yoast/Tests/NamingConventions/ValidHookNameUnitTest.inc

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * Verify that the original functionality of the upstream sniff is still working as expected.
+ * Functionality when no prefixes have been passed, should be unchanged.
+ */
+
+do_action( $Hook_name ); // OK.
+do_action( "admin_head" ); // OK.
+do_action( 'admin_head_' .  get_ID() . '_action' ); // OK.
+do_action( "admin_head_$Post" ); // OK.
+do_action( 'prefix_block_' . $block['blockname'] ); // OK.
+
+do_action( "adminHead" ); // Error - use lowercase.
+do_action( 'admin_Head_' . $Type . '_Action' ); // Error - use lowercase.
+do_action( "admin_head-$hook_suffix" ); // Warning - use underscore.
+apply_filters( 'adminHead', $variable ); // Error - use lowercase.
+do_action( "admin_Head_{$post->ID}_Action" ); // Error - use lowercase.
+do_action( "admin_Head_{$obj->Values[3]->name}-Action_{$obj->Values[3]->name}_Action" ); // Error - use lowercase + warning about dash.
+
+/*
+ * Verify the Yoast specific changes to allow for the plugin prefix, while still checking
+ * that the rest of the hook name is lowercase and underscore separated.
+ *
+ * @phpcs:disable Yoast.NamingConventions.ValidHookName.WrongPrefix
+ */
+
+// phpcs:set Yoast.NamingConventions.ValidHookName prefixes[] Yoast\WP\Plugin,yoast_plugin
+
+apply_filters_ref_array( 'yoast_plugin_some_hook_name', $var ); // OK.
+apply_filters( 'Yoast\WP\Plugin\some_hook_name', $var ); // OK.
+
+do_action_ref_array( 'Yoast\WP\Plugin\some-hook\name', $var ); // Warning - disallowed word separators (after prefix).
+apply_filters( 'Yoast\WP\Plugin\some_HookName', $var ); // Error - uppercase chars (after prefix).
+apply_filters_ref_array( 'yoast_plugin_Some-Hook-Name', $var ); // Error + warning, uppercase chars + dashes (after prefix).
+
+// OK, ignored. Hooks do not start with correct prefix, while valid prefixes have been passed.
+do_action( 'Yoast\WP\AnotherPlugin\some_hook_name', $var );
+apply_filters( 'some_hook_name_yoast_plugin_Prefix_not-at_start', $var );
+
+// Second use of the prefix (correctly) not taken into account as not found at the start of the hook name.
+do_action( 'Yoast\WP\Plugin\hook_' . $type . 'Yoast\WP\Plugin\hook' ); // Error + warning x 2, prefix repeated, compound name.
+
+// Test handing of compound hook names with the prefix as stand-alone string.
+apply_filters( 'Yoast\WP\Plugin\\' . $type . '_' . $sub, $var ); // Warning at severity 3 compound name.
+
+// Test handling of escaped slash in double quotes string.
+$var = apply_filters( "Yoast\WP\Plugin\\{$obj->prop['type']}_details", $var ); // Warning at severity 3 compound name.
+
+// phpcs:enable
+
+/*
+ * Test Yoast specific sniff additions for checking hook name length and such.
+ *
+ * These checks are only in effect if a prefix is found.
+ * The WPCS PrefixAllGlobals sniff checks that a prefix is used, that's outside the scope of this sniff.
+ */
+
+/*
+ * Simple strings, no prefix. Includes testing handling of comments.
+ */
+do_action(
+	// phpcs:ignore Stnd.Cat.Sniff -- For reasons.
```

Comment:
What is the purpose of this line?

Reply:
This is the test case file for the sniff.

A PHPCS test case file should contain good and bad code to test the sniff.
Also: a sniff cannot presume that other rules within the same ruleset will be active, so should contain as much convoluted code as is necessary to test the sniff properly.

With this particular test case, I'm testing that the sniff will still recognize the hook name correctly, even when there are comments and PHPCS annotations within the same parameter.

PHPCS annotations are tokenized differently from comments, which is why I have both in the test case code.

As I don't want this sniff to actually ignore anything, but just want to test that the sniff handles itself correctly when any PHPCS annotation is found in the parameter, I'm ignoring a non-existent sniff, just to have a PHPCS annotation token in the test case.

Does that explain it well enough ?

Reply:
Exhaustively 😁👍